### PR TITLE
MCR-2027 ModsMerger now handles empty or null string

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2027).

### CODE REVIEW

**Comment by fluetze** [comment]
Please add a test for this in MCRNameMergerTest.java

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on d9f257a3e4b60cf4665c1d08bfdf32609fda575f! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5073-pull_request-883

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/mycore/mycore/pullRequest?prid=4549276&bid=15304532)

File: mycore-mods/src/test/java/org/mycore/mods/merger/MCRNameMergerTest.java [hunk 1]
```diff
@@ -133,6 +133,14 @@ public void testPreferFamilyGiven() throws JaxenException, IOException {
         MCRMergerTest.test(a, b, b);
     }
 
+    @Test
+    public void testEmptyGiven() throws JaxenException, IOException {
```
Comment (adlerre) [inline mycore-mods/src/test/java/org/mycore/mods/merger/MCRNameMergerTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4549276)

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-mods/src/main/java/org/mycore/mods/merger/MCRNameMerger.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/mycore/mycore/pullRequest?prid=4549276&bid=15304532)

File: mycore-mods/src/test/java/org/mycore/mods/merger/MCRNameMergerTest.java [hunk 2]
```diff
@@ -18,17 +18,20 @@
...
 import org.jdom2.Element;
 import org.junit.Assert;
 import org.junit.Test;
+import org.mycore.common.MCRConstants;
 import org.mycore.common.MCRTestCase;
 import org.mycore.common.xml.MCRNodeBuilder;
-
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
+import org.mycore.common.xml.MCRXMLHelper;
```
Comment (adlerre) [inline mycore-mods/src/test/java/org/mycore/mods/merger/MCRNameMergerTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.mycore.common.xml.MCRXMLHelper'](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4549276)

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- mycore-mods/src/main/java/org/mycore/mods/merger/MCRNameMerger.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/mycore/mycore/pullRequest?prid=4549276&bid=15304532)

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-mods/src/main/java/org/mycore/mods/merger/MCRNameMerger.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/mycore/mycore/pullRequest?prid=4549276&bid=15304532)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 2b53771201edd2a3651013fb66869719363144bb! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5093-pull_request-883

**Review by kkrebs — APPROVED** [review]
successfully tested with MIR 2018.06.0.5

**Pull request merged**